WeakAuras/Profiling.lua

```diff
@@ -6,6 +6,132 @@ local profileData = {}
+    if type(v) ~= "userdata" then
+      if type(v) == "table" then
+        v = WeakAuras.table_to_string(v, (depth and depth + 1 or 1))
+      elseif type(v) == "function" then
+        v = "function"
+      elseif type(v) == "string" then
+        v = '"'.. v ..'"'
+      end
+
+      if type(k) == "string" then
+        k = '"' .. k ..'"'
+      end
+
+      str = (str and str .. "|cff999999,|r " or "|cff999999{|r ") .. "|cffffff99[" .. tostring(k) .. "]|r |cff999999=|r |cffffffff" .. tostring(v) .. "|r"
+    end
+  end
+  return (str or "{ ") .. " }"
+end
+
+local debugpopup
+local function CreateDebugPopup()
+  local p = CreateFrame("EditBox", "WADebugEditBox", UIParent)
+  p:SetFrameStrata("DIALOG")
+  p:SetMultiLine(true)
+  p:SetAutoFocus(true)
+  p:SetFontObject(ChatFontNormal)
+  p:SetSize(450, 300)
+  p:Hide()
+
+  p.orig_Hide = p.Hide
+  function p:Hide()
+    self:SetText("")
+    self.ScrollFrame:Hide()
+    self.Background:Hide()
+    self:orig_Hide()
+  end
+
+  p.orig_Show = p.Show
+  function p:Show()
+    self.ScrollFrame:Show()
+    self.Background:Show()
+    self:orig_Show()
+  end
+
+  function p:AddText(v)
+    if not v then return end
+    local m = p:GetText()
+    if m ~= "" then
+      m = m .. "|n"
+    end
+    if type(v) == "table" then
+      v = WeakAuras.table_to_string(v)
+    end
+    p:SetText(m .. v)
+  end
+
+  p:SetScript("OnEscapePressed", function(self)
+    self:ClearFocus()
+    self:Hide()
+    self.ScrollFrame:Hide()
+    self.Background:Hide()
+  end)
+
+  local s = CreateFrame("ScrollFrame", "WADebugEditBoxScrollFrame", UIParent, "UIPanelScrollFrameTemplate")
+  s:SetMovable(true)
+  s:SetFrameStrata("DIALOG")
+  s:SetSize(450, 300)
+  s:SetPoint("CENTER")
+  s:SetHitRectInsets(-4, -4, -4, -4)
+  s:SetScrollChild(p)
+  s:Hide()
+
+  s:SetScript("OnMouseDown", function(self, button)
+    if button == "RightButton" and not self.is_moving then
+      self:StartMoving()
+      self.is_moving = true
+    elseif button == "LeftButton" then
+      self:GetScrollChild():SetFocus()
+    end
+  end)
+
+  s:SetScript("OnMouseUp", function(self, button)
+    if button == "RightButton" and self.is_moving then
+      self:StopMovingOrSizing()
+      self.is_moving = nil
+    end
+  end)
+
+  local bg = CreateFrame("Frame", nil, UIParent)
+  bg:SetFrameStrata("DIALOG")
+  bg:SetBackdrop({
+    bgFile = "Interface\\ChatFrame\\ChatFrameBackground",
+    edgeFile = "Interface\\Tooltips\\UI-Tooltip-border",
+    edgeSize = 16,
+    insets = { left = 4, right = 4, top = 4, bottom = 4 }
+  })
+  bg:SetBackdropColor(.05, .05, .05, .8)
+  bg:SetBackdropBorderColor(.5, .5, .5)
+  bg:SetPoint("TOPLEFT", s, -10, 10)
+  bg:SetPoint("BOTTOMRIGHT", s, 30, -10)
+  bg:Hide()
+
+  p.ScrollFrame = s
+  p.Background = bg
+
+  debugpopup = p
+end
+
+WeakAuras.DebugPopup = function()
+  -- create/get and return reference to debug EditBox
+  if not debugpopup then
+    CreateDebugPopup()
+  end
+
+  debugpopup:Hide()
```

Comment:
That's a bit strange place to call hide.

Reply:
Why exactly? Also remember, I took this directly from KNP.